dict_tools/__init__.py

```diff
@@ -7,6 +7,29 @@ def map_keys(function, dictionary):
 
 def map_values(function, dictionary):
     return {
-        key: function(value),
+        key: function(value)
+        for key, value in dictionary.items()
+    }
+
+
+def map_all(function, dictionary):
+    return {
+        function(key): function(value)
+        for key, value in dictionary.items()
+    }
+
+
+def iter_multi_values(dictionary):
```

Comment:
This already exists in more-itertools, I'd rather not duplicate their work.

Reply:
What is the function's name in more-itertools?

Reply:
flatten. It's not exactly the same, but it's close enough that I don't think this specific thing is useful enough to keep.

Reply:
I don't see how flatten is similar to iter_multi_values, which is useful for having multiple values for the same key in a dictionary and iterating through them separately.

Given this dictionary:
```python
{
    'fbi': ('open', 'up'),
    'hotel': 'trivago',
}
```

iter_multi_values gives this result:
```python
[
    ('fbi', 'open'),
    ('fbi', 'up'),
    ('hotel', 'trivago'),
]
```

Whereas flatten in more-itertools operates on a list, so this list:
```python
[['fbi', 'open'], ['up']]
```

gives this result:
```python
['fbi', 'open', 'up']
```

These two functions don't solve the same problem.

Reply:
Ah, I misread it, thanks for the explanation. Given my misunderstanding, we probably need a better name. I'm thinking maybe "flatten_by_values"?

Reply:
Maybe it should be iter_flattened_values

Reply:
Or flattened_items

Reply:
Yeah I'll use flattened_items